src/Server/Helper.php

```diff
@@ -499,6 +499,10 @@ public function parsePsrRequest(ServerRequestInterface $request)
                         Utils::printSafeJson($bodyParams)
                     );
                 }
+
+                if (empty($bodyParams)) {
+                    $bodyParams = json_decode($request->getBody(), true);
+                }
```

Comment:
Empty is a bit too loose. I guess we should check for null only. Thoughts?

Reply:
This is intended, as mentioned in commit message a null value would mean no body at all:

> A null value indicates the absence of body content.

http://www.php-fig.org/psr/psr-7/#321-psrhttpmessageserverrequestinterface

So we test for an empty array, to try to parse the body ourselves. But if it was null it should throw an exception. IIRC the null case is correctly covered by pre-existing tests.

Reply:
Yeah, you are right